### PR TITLE
Bot obfuscation capabilities #93 - Part 1 additional searches over minimum

### PR DESCRIPTION
**Pull request by bbenz3**
Added variables to increase number of searches over the minimum to
decrease chances of being identified as a bot.

### CODE REVIEW

**Comment by bbenz3** [comment]
So I think I have finally completed the changes properly and it is ready for a merge. I accidentally messed with the onComplete.sh file which you can simply ignore (I did revert to the previous fix).

File: main.py [hunk 1]
```diff
@@ -225,13 +226,13 @@ def __run(config):
...
+        print "=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-= FINAL REPORT =-=-=-=-=-=-=-=-=-=-=-=-=-=-=-=-="
         print
-        print "          Account          | Before | After  | Earned | Retries | Lifetime Credits"
-        print "---------------------------+--------+--------+--------+---------+-----------------"
+        print "            Account           | Before | After  | Earned |  Lifetime  | Retries "
+        print "------------------------------+--------+--------+--------+------------+---------"
 
         for r in report:
-            print " %25s | %6d | %6d | %6d | %7d | %16d" % (__stringifyAccount(r, 25), r.oldPoints, r.newPoints, r.pointsEarnedRetrying, r.retries, r.lifetimeCredits)
+            print " %-28s | %6d | %6d | %6d | %10d | %7d" % (__stringifyAccount(r, 28), r.oldPoints, r.newPoints, r.pointsEarnedRetrying, r.lifetimeCredits, r.retries)
```
Comment (sealemar) [inline main.py]: out of curiosity, why did you change account alignment to the right and why the numbers are different now?

File: pkg/bingFlyoutParser.py [hunk 2]
```diff
@@ -59,9 +59,10 @@ class Col:
         RE_EARNED_CREDITS    = (9,    re.compile("Earned \d+ credits?"),   None, True,  Action.PASS)
         COMPLETED            = (10,    "Completed",                        None, False, Action.PASS)
         SILVER_STATUS        = (11,   "Silver Status",                     None, False, Action.PASS)
+        INVITE_FRIENDS       = (12,   "Invite friends",                    None, False, Action.PASS)
```
Comment (sealemar) [inline pkg/bingFlyoutParser.py]: why? you're not referencing it from anywhere

File: pkg/bingFlyoutParser.py [hunk 3]
```diff
@@ -230,7 +231,7 @@ def handle_data(self, data):
                 else:
                     progress = data.strip().split(' of ', 1)
                     self.reward.progressCurrent = int(progress[0])
-                    self.reward.progressMax = int(progress[1])
+                    self.reward.progressMax = int(progress[1].split()[0])
```
Comment (sealemar) [inline pkg/bingFlyoutParser.py]: why split()?

File: version.txt [hunk 4]
```diff
@@ -1,4 +1,25 @@
-Current version 3.8
+Current version 3.8.7
+
+3.8.7   *) Bingbot obfuscation additions
+           
+		   Added the ability to perform additional searches with salted values.
```
Comment (sealemar) [inline version.txt]: please, have indentation correctly.

File: version.txt [hunk 5]
```diff
@@ -1,4 +1,25 @@
...
+		   config.xml has new variables to specify the number of extra searches to perform.
```
Comment (sealemar) [inline version.txt]: please, list those variables here, since otherwise it's a sacred knowledge

File: version.txt [hunk 6]
```diff
@@ -1,4 +1,25 @@
...
+
+3.8.6   *) Adjust Terminal Output To Better Fit 80 Width Terminals
```
Comment (sealemar) [inline version.txt]: why these changes? The **origin** is currently at ```3.8.6```. This is the current [HEAD](https://github.com/sealemar/BingRewards/commit/219b789afc9c55207533b06dfc31658257026d4f)

Comment (amayer5125) [inline version.txt]: It looks like @bbenz3 needs to rebase against our current master. That would get rid of a lot of these changes.

File: pkg/config.py [hunk 7]
```diff
@@ -51,6 +51,10 @@ def __init__(self):
             self.betweenQueriesSalt      = 3.0    # default to this number of seconds
             self.betweenAccountsInterval = 30.0   # default to this number of seconds
             self.betweenAccountsSalt     = 35.5   # default to this number of seconds
+            self.addSearchesDesktop      = 1      # default to this number of additional Desktop searches
+            self.addSearchesDesktopSalt  = 1      # default to this number of additional Desktop searches for salt
+            self.addSearchesMobile       = 1      # default to this number of additional Mobile searches
+            self.addSearchesMobileSalt   = 1      # default to this number of additional Mobile searches for salt
```
Comment (amayer5125) [inline pkg/config.py]: Shouldnt these be 0? So if the user doesnt specify any additional searches it doesnt do any additional searches?

File: pkg/config.py [hunk 8]
```diff
@@ -51,6 +51,10 @@ def __init__(self):
...
```
Comment (amayer5125) [inline pkg/config.py]: Shouldnt these defaults be 0? So if the user doesnt specify any additional searches it doesnt do any additional searches?

Comment (sealemar) [inline pkg/config.py]: I second that

**Comment by Dmurph24** [comment]
Is this still being developed? Its something that will help the script a lot. Im surprised it didnt already have search count salts.

**Comment by sealemar** [comment]
That hasn't been merged yet. The most recent comments haven't been resolved.

**Comment by PlasmaPower** [comment]
@bbenz3 Would you mind if I made my own fork of this and made a seperate PR with these issues fixed?

File: pkg/bingFlyoutParser.py (hunk 3 above)
Comment (PlasmaPower) [inline pkg/bingFlyoutParser.py]: @sealemar It looks like this is the fix from #104, but this still shows up as a change since it is not based off of that commit.

Comment (sealemar) [inline pkg/bingFlyoutParser.py]: @PlasmaPower probably. This PR looks outdated. I think we'll need a new one

**Comment by bbenz3** [comment]
I would have no issue with this. I'm sorry that I haven't had much time to fix this issue but work has been rather crazy lately. I am pretty sure that it works, at least it has been working on my local accounts so if you clean up all of the old modifications then it shouldn't take much to complete it.

**Comment by amayer5125** [comment]
I believe this just needs rebased and the 1 or 2 things @sealemar suggested. If you rebase and push we can review the newer version.

**Comment by sealemar** [comment]
Merged in #113
